### PR TITLE
ENG-970 clear group list before adding the new ones

### PR DESCRIPTION
**Pull request by firegloves**
Fixed EntityDto clearing the group list before adding new groups
Please note that I'm updating a generic class. It may impact other parts of code, but the bug that I'm fixing seems to be a bug in all possible scenarios IMHO

### CODE REVIEW

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **ampie**

If they are not already assigned, you can assign the PR to them by writing `/assign @ampie` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-engine).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/entando-engine/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/entando-engine/pull/21#pullrequestreview-459280148" title="Approved">Kerruba</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **ampie**

If they are not already assigned, you can assign the PR to them by writing `/assign @ampie` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-engine).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/entando-engine/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: src/main/java/org/entando/entando/aps/system/services/entity/model/EntityDto.java [hunk 1]
```diff
@@ -73,6 +73,7 @@ public void fillEntity(IApsEntity prototype, ICategoryManager categoryManager, B
         prototype.setDescription(getDescription() == null ? prototype.getDescription() : getDescription());
         prototype.setMainGroup(getMainGroup() == null ? prototype.getMainGroup() : getMainGroup());
         if (null != this.getGroups()) {
+            prototype.getGroups().clear();
```
Comment (joewhite101) [inline src/main/java/org/entando/entando/aps/system/services/entity/model/EntityDto.java:76]: Add a `setGroups` method and avoid the 2 step invocation? Minor optimization

Comment (firegloves) [reply src/main/java/org/entando/entando/aps/system/services/entity/model/EntityDto.java]: It seems wise, updated

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/entando/entando-engine/pull/21#pullrequestreview-459328673" title="Approved">joewhite101</a>*, *<a href="https://github.com/entando/entando-engine/pull/21#pullrequestreview-459280148" title="Approved">Kerruba</a>*
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approver: **ampie**

If they are not already assigned, you can assign the PR to them by writing `/assign @ampie` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-engine).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/entando-engine/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by entando-jx** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

Approval requirements bypassed by manually added approval.

This pull-request has been approved by: *<a href="https://github.com/entando/entando-engine/pull/21#pullrequestreview-459328673" title="Approved">joewhite101</a>*, *<a href="https://github.com/entando/entando-engine/pull/21#pullrequestreview-459280148" title="Approved">Kerruba</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=entando%2Fentando-engine).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/entando/entando-engine/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**